homework/linux.txt

```diff
@@ -0,0 +1,48 @@
+105:root         703  0.0  1.1 129012 22704 ?        Ssl  12:02   0:00 /usr/bin/python3 /usr/share/unattended-upgrades/unattended-upgrade-shutdown --wait-for-signal
+106:whoopsie     738  0.0  0.7 253128 15556 ?        Ssl  12:02   0:00 /usr/bin/whoopsie -f
+114:konrad       942  0.1  0.9 1943708 18840 ?       S<sl 12:02   0:00 /usr/bin/pulseaudio --daemonize=no --log-target=journal
+116:konrad       947  0.0  0.3 251316  6972 ?        Sl   12:02   0:00 /usr/bin/gnome-keyring-daemon --daemonize --login
+117:konrad       951  0.0  0.3 175180  6624 tty2     Ssl+ 12:02   0:00 /usr/lib/gdm3/gdm-x-session --run-script env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --systemd --session=ubuntu
+120:konrad       959  0.0  0.3   9236  6440 ?        Ss   12:02   0:00 /usr/bin/dbus-daemon --session --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
+132:konrad      1122  0.0  0.0  33772   360 ?        S    12:02   0:00 /usr/bin/VBoxClient --clipboard
+133:konrad      1123  0.0  0.1 165908  3996 ?        Sl   12:02   0:00 /usr/bin/VBoxClient --clipboard
+134:konrad      1131  0.0  0.0  33772   360 ?        S    12:02   0:00 /usr/bin/VBoxClient --seamless
+135:konrad      1133  0.0  0.1 166008  2740 ?        Sl   12:02   0:00 /usr/bin/VBoxClient --seamless
+136:konrad      1136  0.0  0.0  33772   364 ?        S    12:02   0:00 /usr/bin/VBoxClient --draganddrop
+137:konrad      1138  0.0  0.1 166524  2716 ?        Sl   12:02   0:00 /usr/bin/VBoxClient --draganddrop
+138:konrad      1146  0.0  0.0  33772   360 ?        S    12:02   0:00 /usr/bin/VBoxClient --vmsvga
+139:konrad      1147  0.0  0.1 168148  3428 ?        Sl   12:02   0:00 /usr/bin/VBoxClient --vmsvga
+140:konrad      1156  0.0  0.0   6040   452 ?        Ss   12:02   0:00 /usr/bin/ssh-agent /usr/bin/im-launch env GNOME_SHELL_SESSION_MODE=ubuntu /usr/bin/gnome-session --systemd --session=ubuntu
+142:konrad      1178  0.0  0.2   7248  4296 ?        S    12:02   0:00 /usr/bin/dbus-daemon --config-file=/usr/share/defaults/at-spi2/accessibility.conf --nofork --print-address 3
+145:konrad      1207  1.5 11.2 4059904 228356 ?      Ssl  12:02   0:06 /usr/bin/gnome-shell
+157:konrad      1294  0.0  1.3 2609700 26532 ?       Sl   12:02   0:00 /usr/bin/gjs /usr/share/gnome-shell/org.gnome.Shell.Notifications
+182:konrad      1561  0.3  3.5 1470596 73224 ?       Sl   12:03   0:01 /usr/bin/nautilus --gapplication-service
+187:konrad      1896  0.7  2.7 829356 56904 ?        Sl   12:07   0:00 /usr/bin/gedit --gapplication-service
+  304  mkdir Homework
+  305  cd Homework/
+  306  touch processes.txt
+  307  ps aux > processes.txt 
+  308  cat processes.txt 
+  309  mkdir backup
+  310  touch copy.txt | cp processes.txt copy.txt
+  311  mv copy.txt backup/
+  312  touch selected.txt | wc -l > selected.txt
+  313  cat selected.txt 
+  314  touch selected.txt | wc -l processes.txt > selected.txt
+  315  cat selected.txt 
+  316  grep -rni "/bin" processes.txt >> selected.txt 

```

Comment:
-r is not needed because it means you want to grep recurent. Here we just check one file ;)